src/solver/adaptive_time_step.cpp

```diff
@@ -114,8 +114,17 @@ void adaptive_time_step::parse_input(json const& increment_data, double const ma
     // Initial factor determined by input
     initial_time = increment_data["Increments"]["Initial"];
 
-    minimum_increment = increment_data["Increments"]["Minimum"];
-    maximum_increment = increment_data["Increments"]["Maximum"];
+    auto adaptive_increment = increment_data["Increments"]["Adaptive"];
```

Comment:
I like this idea.  The JSON spec allows boolean types, so to avoid a string allocation and compare we can just do:
```cpp
bool const is_adaptive_increment = increment_data["Increments"]["Adaptive"];

minimum_increment = is_adaptive_increment ? increment_data["Increments"]["Minimum"] : initial_time;
...
```
Since only two possibilities are likely.  The JSON would look like
```json
...
"Adaptive" : false,
...
```